Fixing exceptions when attribute contains no value

Chopinsky opened the pull request:
This PR is to address the exception being thrown if the target html file contains `link` tags whose `type` attribute is defined but without a value. 

For example, when using the plugin with webpack, encountering the following `link` tag: 

`<link href="https://somecdn.com/favicon/favicon.png" rel="icon" type>` 

in a html file will crash the build process because of this exception: `TypeError: Cannot read property 'toLowerCase' of undefined`.

With this fix, we will treat the undefined attribute value as a blank string value, i.e. it will be equivalent as the `link` tag below:

`<link href="https://somecdn.com/favicon/favicon.png" rel="icon" type="">`.

iastate-zuo committed:
fixing exceptions when attribute contains no value

iastate-zuo committed:
handling other illegal entries as well

DanielRuf commented on src/htmlminifier.js:
```diff
@@ -182,7 +182,14 @@ function isExecutableScript(tag, attrs) {
-  attrValue = trimWhitespace(attrValue).toLowerCase();
+  attrValue = trimWhitespace(attrValue);
+
+  if (typeof attrValue !== 'string') {
```
Wouldn't be `typeof attrValue !== 'undefined'` be better?

Because the quotes are often optional (unquoted attribute values) so this would change `val=integer` to `val=''`.

DanielRuf replied:
https://dev.w3.org/html5/html-author/Overview.src.html

> There are four slightly different syntaxes that may be used for attributes in HTML: empty, unquoted, single-quoted and double-quoted. All four syntaxes may be used in the HTML syntax, depending on what is needed for each specific attribute. However, in the XHTML syntax, attribute values must always be quoted using either single or double quotes.

https://mathiasbynens.be/notes/unquoted-attribute-values

Only undefined / empty values should be treated this way.

Chopinsky replied:
I've made the update to only handle the 'undefined' values here.

Chopinsky committed:
only handle exceptions with 'undefined' values

reverting from a more sweeping change where any attrValue not of the type of 'string' is treated as a blank entry.
